server/src/types.ts

```diff
@@ -10,11 +10,25 @@ export type DoenetmlVersion = {
 
 export type AssignmentStatus = "Unassigned" | "Closed" | "Open";
 
+export type LibraryInfo = {
+  sourceId: Uint8Array;
+  activityId: Uint8Array | null;
+  ownerRequested?: boolean;
+  status:
+    | "none"
+    | "PENDING_REVIEW"
+    | "REQUEST_REMOVED"
+    | "PUBLISHED"
+    | "NEEDS_REVISION";
+  comments?: string;
+};
+
 export type UserInfo = {
   userId: Uint8Array;
   firstNames: string | null;
   lastNames: string;
   email: string;
+  // isLibrary: boolean;
```

Comment:
We could leave `isLibrary` on the `UserInfo` on server, then modify `userConvertUUID` to strip it off before sending to client. (Might need to change the name of `userConvertUUID`.) Then `model` doesn't need to worry about removing the library and as it will always be done before sending it on to the client.